js/modules/pattern-fill.src.js

```diff
@@ -597,7 +599,7 @@ H.Chart.prototype.callbacks.push(function (chart) {
         'M 0 0 L 5 10 L 10 0'
     ].forEach(function (pattern, i) {
         chart.renderer.addPattern({
-            id: 'highcharts-default-pattern-' + i,
+            id: 'highcharts-default-pattern-' + (index ? index + '-' : '') + i,
```

Comment:
Can index equal `0` in this case? If so then you should probably test for `isNumber(index) ? : ;` instead.

Reply:
It can, but on purpose we don't want to append the index to the ID in those cases. This way we won't break backwards compatibility